pgmpy/tests/test_models/test_DynamicBayesianNetwork.py

```diff
@@ -137,6 +137,12 @@ def test_moralize(self):
                               [('D', 1), ('G', 1)], [('D', 1), ('I', 1)],
                               [('G', 0), ('I', 0)], [('G', 1), ('I', 1)]])
 
+    def test_copy(self):
+        self.network_copy = self.network.copy()
+        self.assertListEqual(self.network.nodes(), self.network_copy.nodes())
+        self.assertListEqual(self.network.edges(), self.network_copy.edges())
+        self.assertListEqual(self.network.get_cpds(), self.network_copy.get_cpds())
+       
```

Comment:
@pdoongarwal  Add more test to check for mutability and related issues. Like changing something in copy doesn't changes things in original one. Look at the other PR for copy method and see how they are doing this.